miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppMessageBridge.kt

```diff
@@ -62,8 +62,21 @@ abstract class MiniAppMessageBridge {
         permissionsWithDescription: List<Pair<MiniAppCustomPermissionType, String>>,
         callback: (List<Pair<MiniAppCustomPermissionType, MiniAppCustomPermissionResult>>) -> Unit
     ) {
-        throw MiniAppSdkException("The `MiniAppMessageBridge.requestCustomPermissions`" +
-                " method has not been implemented by the Host App.")
+        throw MiniAppSdkException(
+            "The `MiniAppMessageBridge.requestCustomPermissions`" +
+                    " method has not been implemented by the Host App."
+        )
+    }
+
+    /**
+     * Post user name request from external.
+     * @param callback to return user name.
+     */
+    open fun requestUserName(callback: (isSuccess: Boolean, data: String) -> Unit) {
```

Comment:
I think the function name should be `getUserName` - this matches the naming convention of `getUniqueId`.

Also, I'm not sure if we need to use a `callback` here. I was thinking this function would be the same as `getUniqueId` and just directly return a `String`. What do you think @khairul-alam-licon @minh-rakuten ?

Reply:
It looks like iOS is taking the same approach of having a callback: https://github.com/rakutentech/ios-miniapp/pull/149/files#diff-eeb7f9f814d262c7e3820df82237603bR6

But on iOS, they have a `Result` with `success` and `failure` callbacks. Looks like Kotlin also has a `Result` class now, so maybe we should try that? https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/#result

Reply:
Directly return a string is fine to me. We should simplify the usage for the developers in host app.

Reply:
Thanks for the suggestion. I liked the concept of using [Result](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/#result). 
Firstly, I wanted to do `fun getUserName(): Result<String>` but `Result` can't be used as a return type unfortunately. 

So, I went with `fun requestUserName(callback: (result: Result<String>) -> Unit)`. But, at the end I think, we don't need to handle those `success` or `failure` in HostApp, rather we can handle everything inside SDK, let users to use this feature easily.

So, I am preferring finally `abstract fun getUserName(): String`. I think, @minh-rakuten already has a `+1` on it.
@corycaywood What do you think?

Reply:
@khairul-alam-licon Yes, I think `fun getUserName(): String` is okay - it's best to be consistent among the functions that have similar functionality (`getUniqueId`).

@ts-leojoseph-r I think you implemented a callback for this on iOS. Can it be updated to directly return a string instead?

Reply:
Sure @corycaywood  👍 . Initial plan I had was to return some more custom errors from the Host app, I guess just returning the string would suffice now for both Username and Profile Photo.